src/Service.Tests/CosmosTests/SchemaGeneratorTest.cs

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using Azure.DataApiBuilder.Config;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Generator;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Tests.CosmosTests
+{
+    /// <summary>
+    /// Contains unit tests for the <see cref="SchemaGenerator"/> class to verify its schema generation capabilities 
+    /// based on various JSON data inputs. This class ensures that the schema generation logic is accurate and 
+    /// robust for different data structures and configurations.
+    /// </summary>
+    [TestClass, TestCategory(TestCategory.COSMOSDBNOSQL)]
+    public class SchemaGeneratorTest
+    {
+        /// <summary>
+        /// Tests the <see cref="SchemaGenerator.Generate"/> method using a single JSON file to generate a GraphQL schema.
+        /// Verifies that the generated schema matches the expected schema output.
+        /// </summary>
+        /// <param name="jsonFilePath">Path to the directory containing the input JSON file.</param>
+        /// <param name="gqlFilePath">Path to the directory containing the expected GraphQL schema file.</param>
+        [TestMethod]
+        [DataRow("CosmosTests/TestData/CosmosData", "CosmosTests/TestData/GeneratedGqlSchema")]
+        public void TestSchemaGenerator(string jsonFilePath, string gqlFilePath)
+        {
+            string json = Regex.Replace(File.ReadAllText($"{jsonFilePath}/EmulatorData.json", Encoding.UTF8), @"\s+", string.Empty);
+            List<JsonDocument> jsonArray = new() { JsonSerializer.Deserialize<JsonDocument>(json) };
+
+            string actualSchema = SchemaGenerator.Generate(jsonArray, "containerName");
+            string expectedSchema = File.ReadAllText($"{gqlFilePath}/EmulatorData.gql");
+
+            AreEqualAfterCleanup(expectedSchema, actualSchema);
+        }
+
+        /// <summary>
+        /// Tests the <see cref="SchemaGenerator.Generate"/> method using multiple JSON files to generate a GraphQL schema.
```

Comment:
what do you mean by multiple json files? multiple DAB runtime configs? multiple json structures in cosmosdb?

Reply:
imitating multiple records in cosmos DB. In Cosmos DB, each record can have different schema.